packages/mui-base/src/SwitchUnstyled/SwitchUnstyled.tsx

```diff
@@ -76,48 +73,45 @@ const SwitchUnstyled = React.forwardRef(function SwitchUnstyled(
   };
 
   const Root: React.ElementType = component ?? components.Root ?? 'span';
-  const rootComponentProps = resolveComponentProps(componentsProps.root, ownerState);
-  const rootProps: WithOptionalOwnerState<SwitchUnstyledRootSlotProps> = appendOwnerState(
-    Root,
-    {
-      ...otherProps,
-      ...rootComponentProps,
-      className: clsx(classes.root, stateClasses, className, rootComponentProps?.className),
+  const rootProps: WithOptionalOwnerState<SwitchUnstyledRootSlotProps> = useSlotProps({
+    elementType: Root,
+    externalSlotProps: componentsProps.root,
+    externalForwardedProps: otherProps,
+    additionalProps: {
+      ref,
     },
     ownerState,
-  );
+    className: [classes.root, stateClasses, className],
```

Comment:
In some components we are descructuring the `className` prop and passing it here, and in some not. Is there a specific reason for it? If not, let's keep it consistent.

Reply:
This applies to other PRs too :)

Reply:
I realized today while working with Badge that it is not necessary. I'll keep it consistent and change all the remaining occurences.